File: backend/src/FieldTicket.Shared/Models/VerificationModels.cs
using System.Text.Json;

namespace FieldTicket.Shared.Models;

/// <summary>
/// 提交验证结果请求
/// </summary>
public class SubmitVerificationRequest
{
    /// <summary>
    /// 关联的解决方案ID（可选）
    /// </summary>
    public Guid? SolutionId { get; set; }
    
    /// <summary>
    /// 验证次数
    /// </summary>
    public int RunCount { get; set; }
    
    /// <summary>
    /// 通过次数
    /// </summary>
    public int PassCount { get; set; }
    
    /// <summary>
    /// 失败次数
    /// </summary>
    public int FailCount { get; set; }
    
    /// <summary>
    /// 验证清单结果（JSON格式）
    /// </summary>
    public JsonDocument ChecklistResultJson { get; set; } = JsonDocument.Parse("{}");
    
    /// <summary>
    /// 证据附件ID列表
    /// </summary>
    public List<Guid> EvidenceAttachmentIds { get; set; } = new();
    
    /// <summary>
    /// 验证备注
    /// </summary>
    public string? Note { get; set; }
}

/// <summary>
/// 验证结果DTO
/// </summary>
public class VerificationDto
{
    public Guid VerificationId { get; set; }
    public Guid TicketId { get; set; }
    public Guid? SolutionId { get; set; }
    public Guid ExecutedBy { get; set; }
    public string? ExecutedByName { get; set; }
    
    public int RunCount { get; set; }
    public int PassCount { get; set; }
    public int FailCount { get; set; }
    
    public string Result { get; set; } = string.Empty;
    
    public JsonDocument ChecklistResultJson { get; set; } = JsonDocument.Parse("{}");
    public List<Guid> EvidenceAttachmentIds { get; set; } = new();
    
    public string? Note { get; set; }
    
    public DateTime VerifiedAt { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: mobile-app/lib/models/verification_models.dart
/// 验证相关数据模型

/// 提交验证结果请求
class SubmitVerificationRequest {
  final String? solutionId;
  final int runCount;
  final int passCount;
  final int failCount;
  final Map<String, dynamic> checklistResult;
  final List<String> evidenceAttachmentIds;
  final String? note;

  SubmitVerificationRequest({
    this.solutionId,
    required this.runCount,
    required this.passCount,
    required this.failCount,
    this.checklistResult = const {},
    this.evidenceAttachmentIds = const [],
    this.note,
  });

  Map<String, dynamic> toJson() {
    return {
      if (solutionId != null) 'solutionId': solutionId,
      'runCount': runCount,
      'passCount': passCount,
      'failCount': failCount,
      'checklistResultJson': checklistResult,
      'evidenceAttachmentIds': evidenceAttachmentIds,
      if (note != null && note!.isNotEmpty) 'note': note,
    };
  }
}

/// 验证结果DTO
class VerificationDto {
  final String verificationId;
  final String ticketId;
  final String? solutionId;
  final String executedBy;
  final String? executedByName;
  final int runCount;
  final int passCount;
  final int failCount;
  final String result;
  final Map<String, dynamic> checklistResult;
  final List<String> evidenceAttachmentIds;
  final String? note;
  final DateTime verifiedAt;
  final DateTime createdAt;
  final DateTime updatedAt;

  VerificationDto({
    required this.verificationId,
    required this.ticketId,
    this.solutionId,
    required this.executedBy,
    this.executedByName,
    required this.runCount,
    required this.passCount,
    required this.failCount,
    required this.result,
    this.checklistResult = const {},
    this.evidenceAttachmentIds = const [],
    this.note,
    required this.verifiedAt,
    required this.createdAt,
    required this.updatedAt,
  });

  factory VerificationDto.fromJson(Map<String, dynamic> json) {
    return VerificationDto(
      verificationId: json['verificationId'] as String,
      ticketId: json['ticketId'] as String,
      solutionId: json['solutionId'] as String?,
      executedBy: json['executedBy'] as String,
      executedByName: json['executedByName'] as String?,
      runCount: json['runCount'] as int,
      passCount: json['passCount'] as int,
      failCount: json['failCount'] as int,
      result: json['result'] as String,
      checklistResult: json['checklistResultJson'] as Map<String, dynamic>? ?? {},
      evidenceAttachmentIds: (json['evidenceAttachmentIds'] as List<dynamic>?)
              ?.map((e) => e as String)
              .toList() ??
          [],
      note: json['note'] as String?,
      verifiedAt: DateTime.parse(json['verifiedAt'] as String),
      createdAt: DateTime.parse(json['createdAt'] as String),
      updatedAt: DateTime.parse(json['updatedAt'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'verificationId': verificationId,
      'ticketId': ticketId,
      if (solutionId != null) 'solutionId': solutionId,
      'executedBy': executedBy,
      if (executedByName != null) 'executedByName': executedByName,
      'runCount': runCount,
      'passCount': passCount,
      'failCount': failCount,
      'result': result,
      'checklistResultJson': checklistResult,
      'evidenceAttachmentIds': evidenceAttachmentIds,
      if (note != null) 'note': note,
      'verifiedAt': verifiedAt.toIso8601String(),
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
    };
  }
}
